aicsimageio/transforms.py

```diff
@@ -141,7 +140,7 @@ def reshape_data(
     ... example = reshape_data(data, "CYX", "BSTCZYX", C=slice(0, -1, 3))
     """
     # Check for parameter conflicts
-    for dim in DEFAULT_DIMENSION_ORDER_LIST:
+    for dim in given_dims:
```

Comment:
Random bug that I found that surprisingly never got caught because in the case that a dim wasn't in our defaults, validation of that dim was just ignored. So now this is actually validating all dims passed!

Reply:
I think that may have been intentional whenever we started on this but I think now this makes a lot more sense to handle the way you have.